### PR TITLE
Fix version of puppet

### LINKED ISSUE #2: Errors installing ruby gems
I ran into this issue/solution:

http://stackoverflow.com/questions/22717428/vagrant-error-failed-to-mount-folders-in-linux-guest

Later, when I run this I get some errors trying to install some ruby gems:

```
...
==> default: Updating installed gems
==> default: Updating puppet
==> default: Successfully installed json_pure-1.8.2
==> default: Successfully installed hiera-1.3.4
==> default: Successfully installed puppet-3.7.4
==> default: Gems updated: hiera json_pure puppet
==> default: ERROR:  Error installing librarian-puppet:
==> default: 	i18n requires Ruby version >= 1.9.3.
==> default: Successfully installed multi_json-1.11.0
==> default: Successfully installed multipart-post-2.0.0
==> default: Successfully installed faraday-0.9.1
==> default: Successfully installed thread_safe-0.3.4
==> default: Successfully installed minitest-5.5.1
==> default: Successfully installed tzinfo-1.2.2
==> default: Building native extensions.  This could take a while...
==> default: Successfully installed json-1.8.2
==> default: /tmp/vagrant-shell: 61: /tmp/vagrant-shell: librarian-puppet: not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

@esteve can you try to set this up and see if you get the same problem?

### PR DESCRIPTION
**Pull request by DLu**
Fixes #2 again.

### CODE REVIEW

File: shell/main.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ else
 fi
 
 if [ "$(gem search -i librarian-puppet)" = "false" ]; then
-  gem install librarian-puppet --no-rdoc --no-ri
+  gem install librarian-puppet --no-rdoc --no-ri -v 2.2.3
```
Comment (wjwwood) [inline shell/main.sh]: Can I get a consult on pinning a dependency version in gem @nuclearsandwich? Is this safe/reasonable to do? I don't have much experience with it, and I don't know if you can have more than one version installed at a time (like cargo or npm) or if it pins for the whole system (like pip or homebrew).

Comment (DLu) [reply shell/main.sh]: FWIW, I have no idea how gems or puppet work. But I was getting the error I referenced https://github.com/ros-infrastructure/roswiki_vagrant/issues/2#issuecomment-454508629 and I assumed gem was installing the most recent version (3.0.0) and this was the second most recent version and it gets farther when you do this.

Comment (DLu) [reply shell/main.sh]: Given that I'm getting other errors further down the pipeline, I'd be curious to know what version of the library is currently installed on your VM, @wjwwood

Comment (wjwwood) [reply shell/main.sh]: I haven't used this in years, it's not setup on any of my machines (virtual or otherwise).

Comment (nuclearsandwich) [reply shell/main.sh]: In this instance it's the reasonable thing to do unless/until we update or replace the actual puppet logic. It looks like the 3.0.0 release (Jan 2018) drops support for Ruby 1.x and Puppet 3.7 (There is a puppet 3.8 on Xenial that might work with it but that is a whole extra can of marionette strings).

Comment (wjwwood) [reply shell/main.sh]: Sounds good, thanks for the input!

**Pull request merged**